### DISCUSSION TITLE
lb4 storage and multer component ??

### DISCUSSION THREAD

**Discussion by emazzu** [loopbackio/loopback-next#7358] [Q&A]
Hi, how arew you ??
I would like to know, if in lb4, component storage is replaced by multer, and if with multer I can connect to aws, azure, file system, etc. or now exist distinct connector for each case. 

Regards
Eduardo

**Comment by marioestradarosa** [comment]
@emazzu ,  the Loopback 4 [File transfer example ](https://github.com/strongloop/loopback-next/tree/master/examples/file-transfer) shows how well other packages like [Multer](https://www.npmjs.com/package/multer) can fit in the framework architecture with artifacts like [Service providers](https://loopback.io/doc/en/lb4/Service.html).   

The example exposes a **Multer** _middleware_ thru this provider, hence the storage component and core file handling belongs to **Multer** itself not loopback 4. Having said that, with that example you have the building blocks to integrate other packages related to **Multer** such as [Multer S3](https://www.npmjs.com/package/multer-s3) or [Multer Azure](https://www.npmjs.com/package/multer-azure) and build your application or micro service according to your needs.  

**Loopback4** provides already all the tools with _composability and modularity_ in mind that you can use .
